### PR TITLE
System Test: Module (WAZUH-DB) - Agent forced to change group

### PR DESCRIPTION
**Pull request by CamiRomero**
In this PR the test suite for `Agent forced to change group`  is added. A total of 3 new test cases covering different modes and combinations of options.

<table>
  <tr>
    <th>Related issues</th>
    <th><a href="[url](https://github.com/wazuh/wazuh-qa/issues/2512)">#2512</a></th>
    <th><a href="[url](https://github.com/wazuh/wazuh/issues/10771)">#10771</a></th>
  </tr>
</table>

## Cases:

- [x] Agent forced to change group:
        - The agent  registered in `master` node changed group
        - The agent  registered in `worker` node changed group
        - Force group change twice

--- 

## Configuration options
In order to run the test, first the environment located in `/test/system/provisioning/basic_cluster` must be enabled with:
`sudo ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"wazuh_branch": "dev-10771-agent-groups-files-to-wazuh-db"}'`

**Local Internal Options:** No Local Internal Options used

---

---

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
| test/system/test_cluster/test_agent_groups/test_agent_groups_default.py  | CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8317620/R1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8317840/R1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8317847/R2.zip)| 21/03/2022 |  @CamiRomero  | 

---

## Required
- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

**Comment by Deblintrake09** [comment]
## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
| test/system/test_cluster/test_agent_groups/test_agent_groups_default.py  | CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8248620/R1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8248622/R2.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8248623/R3.zip)| 10/03/2022 |  @CamiRomero  |

**Review by Deblintrake09 — APPROVED** [review]

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                              'provisioning', 'basic_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+
+
+# Tests
+@pytest.mark.parametrize("test_infra_managers",[test_infra_managers])
+@pytest.mark.parametrize("test_infra_agents",[test_infra_agents])
+@pytest.mark.parametrize("host_manager",[host_manager])
+@pytest.mark.parametrize("agent_host", test_infra_managers[0:2])
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:70]: Why is `agent_host `not a list if it contains `test_infra_managers`?

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+def test_force_group_change(agent_host, clean_environment, test_infra_managers, test_infra_agents, host_manager):
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:71]:
```suggestion
def test_sync_when_forced_to_change_a_group(agent_host, clean_environment, test_infra_managers, test_infra_agents, host_manager):
```

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    '''
+    for group in agent_groups:
+        create_new_agent_group(test_infra_managers[0], group, host_manager)
+
+    agent1_data = register_agent(test_infra_agents[0], agent_host, host_manager, agent_groups[0])
+    agent2_data = register_agent(test_infra_agents[1], agent_host, host_manager, agent_groups[1])
+    agent3_data = register_agent(test_infra_agents[2], agent_host, host_manager, agent_groups[2])
+
+    restart_cluster(test_infra_agents, host_manager)
+    time.sleep(10)
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:107]: Remove hardcoded value.

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+
+# Tests
+@pytest.mark.parametrize("test_infra_managers",[test_infra_managers])
+@pytest.mark.parametrize("test_infra_agents",[test_infra_agents])
+@pytest.mark.parametrize("host_manager",[host_manager])
+def test_force_group_change_during_sync(clean_environment, test_infra_managers, test_infra_agents, host_manager):
+    '''
+    description: Check that having a series of agents assigned with different groups, when an new node is added to
+    the cluster, the group data is synchronized to the new node.
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:133]:
```suggestion
    description: Check that having a series of agents assigned with different groups, when an new node is added from the worker node, the group data is synchronized to the new node.
```

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        - agent_host:
+            type: string
+            brief: name of the host where the agent will register
+        - clean_enviroment:
+            type: fixture
+            brief: Reset the wazuh log files at the start of the test. Remove all registered agents from master.
+        - test_infra_managers
+            type: List
+            brief: list of manager hosts in enviroment
+        - test_infra_managers
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:86]:
```suggestion
        - test_infra_agents
```

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    the cluster, the group data is synchronized to the new node.
+    wazuh_min_version: 4.4.0
+    parameters:
+        - clean_enviroment:
+            type: fixture
+            brief: Reset the wazuh log files at the start of the test. Remove all registered agents from master.
+        - test_infra_managers
+            type: List
+            brief: list of manager hosts in enviroment
+        - test_infra_managers
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:142]:
```suggestion
        - test_infra_agents
```

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py (hunk 4 above)
Comment (damarisg) [reply tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:133]: The description is confused and has a lot of similarities with the previous test. Please check it.

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py [hunk 7]
```diff
@@ -0,0 +1,180 @@
+'''
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:1]: I can't see where you connect the agent with the master or with the worker according to cases 1 and 2. We need to do a sync.

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py (hunk 1 above)
Comment (damarisg) [reply tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py:69]: @Deblintrake09

File: tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py (hunk 2 above)
Comment (Deblintrake09) [reply tests/system/test_cluster/test_agent_groups/test_agent_groups_forced_change.py]: Changed

**Review by damarisg — APPROVED** [review]

**Pull request merged**